### PR TITLE
[FIX] sale_payment: Restrain access to order's partners

### PR DESCRIPTION
**Pull request by tbe-odoo**
- Restrain the creation of a payment transaction for an order to he invoice's partner.
- Removed the callback_method argument as it was unused.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/models/sale.py [hunk 1]
```diff
@@ -396,6 +396,7 @@ def action_invoice_create(self, grouped=False, final=False):
         :param final: if True, refunds will be generated if necessary
         :returns: list of created invoices
         """
+        import pudb; pu.db
```
Comment (Yenthe666) [inline addons/sale/models/sale.py]: @tbe-odoo don't forget to remove this before merging it in.

Comment (tbe-odoo) [reply addons/sale/models/sale.py]: Woops committed the wrong thing, thanks :)

Comment (Yenthe666) [reply addons/sale/models/sale.py]: You're welcome, suspected it was a mistake. 😉

**Comment by tbe-odoo** [comment]
Merged here https://github.com/odoo/odoo/commit/f3298859eb99315269296e303a1de6cea4cc5c50

**Pull request closed without merging**